src/pages/iv-materials-tfr/Form/MaterialsTransferForm.js

```diff
@@ -0,0 +1,923 @@
+    receivedDate: null,
+    fromSiteId: null,
+    toSiteId: '',
+    notes: '',
+    status: 1,
+    plantId: parseInt(plantId),
+    printStatus: null,
+    qty: null,
+    pcs: null,
+    isVerified: false,
+    wip: 1,
+    carrierId: null,
+    transfers: [
+      {
+        id: 1,
+        transferId: recordId || 0,
+        seqNo: 1,
+        sku: '',
+        itemName: null,
+        itemId: null,
+        qty: null,
+        componentSeqNo: 0,
+        qtyInBase: 0,
+        muId: null,
+        muQty: '',
+        muRef: '',
+        weight: null,
+        unitCost: 0,
+        msId: null,
+        trackBy: null,
+        lotCategoryId: null,
+        metalId: null,
+        totalCost: 0,
+        priceType: null
+      }
+    ]
+  }
+
+  const invalidate = useInvalidate({
+    endpointId: InventoryRepository.MaterialsTransfer.qry
+  })
+
+  async function getDefaultFromSiteId() {
+    const defaultFromSiteId = userDefaultsData?.list?.find(({ key }) => key === 'siteId')
+    formik.setFieldValue('fromSiteId', parseInt(defaultFromSiteId?.value))
+  }
+
+  const { formik } = useForm({
+    initialValues,
+    maxAccess,
+    enableReinitialize: false,
+    validateOnChange: true,
+    validationSchema: yup.object({
+      date: yup.date().required(),
+      fromSiteId: yup.string().required(),
+      toSiteId: yup.string().required(),
+      transfers: yup
+        .array()
+        .of(
+          yup.object().shape({
+            sku: yup.string().required(),
+            qty: yup.string().required(),
+            msId: yup.string().required()
+          })
+        )
+        .required()
+    }),
+    onSubmit: async values => {
+      const copy = { ...values }
+      delete copy.transfers
+      copy.date = !!copy.date ? formatDateToApi(copy.date) : null
+      copy.closedDate = !!copy.closedDate ? formatDateToApi(copy.closedDate) : null
+      copy.receivedDate = !!copy.receivedDate ? formatDateToApi(copy.receivedDate) : null
+      copy.status = copy.status
+      copy.wip = copy.wip
+
+      const updatedRows = formik?.values?.transfers.map((transferDetail, index) => {
+        return {
+          ...transferDetail,
+          seqNo: index + 1,
+          transferId: formik.values.recordId || 0
+        }
+      })
+      if (values.fromSiteId === values.toSiteId) {
+        stackError({
+          message: labels.errorMessage
+        })
+
+        return
+      }
+
+      const resultObject = {
+        header: copy,
+        items: updatedRows,
+        serials: [],
+        lots: []
+      }
+
+      let notificationData
+
+      const res = await postRequest({
+        extension: InventoryRepository.MaterialsTransfer.set2,
+        record: JSON.stringify(resultObject)
+      })
+
+      if (!values.recordId) {
+        toast.success(platformLabels.Added)
+        formik.setFieldValue('recordId', res.recordId)
+
+        const res2 = await getRequest({
+          extension: InventoryRepository.MaterialsTransfer.get,
```

Comment:
use here refetchForm, prevent code duplication, and remove the separate setters for recordId and reference, as their values will be set directly within this function.